src/plugins/types.ts

```diff
@@ -0,0 +1,6 @@
+import { AnyFunc, WrapWithName } from '../types'
+
+export interface PluginConfig {
+  matchers: WrapWithName<AnyFunc>[]
+  validators: WrapWithName<AnyFunc>[]
```

Comment:
i think that using AnyFunc here makes it harder to write plugins. I would suggest the following signature: `(this: Expectation, ...args: any[]) => void | Promise<void>`. This would enable easier typing for plugin authors.